lib/libzfs/libzfs_dataset.c

```diff
@@ -1249,15 +1256,17 @@ zfs_valid_proplist(libzfs_handle_t *hdl, zfs_type_t type, nvlist_t *nvl,
 			}
 			if (intval != 0 &&
 			    (intval < SPA_MINBLOCKSIZE ||
-			    intval > SPA_OLD_MAXBLOCKSIZE || !ISP2(intval))) {
+			    intval > maxbs || !ISP2(intval))) {
+				zfs_nicebytes(maxbs, buf, sizeof (buf));
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "invalid '%s=%d' property: must be zero or "
-				    "a power of 2 from 512B to 128K"), propname,
-				    intval);
+				    "a power of 2 from 512B to %s"), propname,
```

Comment:
we will want to also update the zfs(8) man page and the property registration. as in:
```
        zprop_register_number(ZFS_PROP_SPECIAL_SMALL_BLOCKS,
            "special_small_blocks", 0, PROP_INHERIT, ZFS_TYPE_FILESYSTEM,
            "zero or 512 to 128K, power of 2", "SPECIAL_SMALL_BLOCKS");
```

Reply:
Good point, fixes.